src/main/java/com/minecolonies/coremod/tileentities/TileEntityColonyBuilding.java

```diff
@@ -262,15 +309,59 @@ public NBTTagCompound writeToNBT(@NotNull final NBTTagCompound compound)
             //todo: actually do something about it and not spam the server
         }
         */
+        compound.setTag(TAG_INVENTORY, itemHandler.serializeNBT());
+        if (customName != null)
+        {
+            compound.setString(TAG_CUSTOM_NAME, customName);
+        }
         compound.setInteger(TAG_COLONY, colonyId);
         compound.setBoolean(TAG_MIRROR, mirror);
         return compound;
     }
 
-    @Override
     public boolean isUseableByPlayer(@NotNull final EntityPlayer player)
     {
-        return super.isUseableByPlayer(player) && this.hasAccessPermission(player);
+        return this.worldObj.getTileEntity(this.pos) != this
+                ? false
+                : (player.getDistanceSq((double)this.pos.getX() + 0.5D, (double)this.pos.getY() + 0.5D, (double)this.pos.getZ() + 0.5D) <= 64.0D
+                    && this.hasAccessPermission(player));
+    }
+
+    /**
+     * Get the name of this object. For players this returns their username
+     */
+    public String getName()
+    {
+        return this.hasCustomName() ? this.customName : "container.chest";
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)